Classes/FrontEnd/DefaultController.php

```diff
@@ -430,6 +445,20 @@ protected function getSingleViewHooks(): array
         return $this->singleViewHooks;
     }
 
+    /**
+     * Gets the hook provider for the single view.
+     *
+     * @return HookProvider
+     */
+    protected function getSingleViewHookProvider(): HookProvider
+    {
+        if (!$this->singleViewHookProvider instanceof HookProvider) {
```

Comment:
Please use `=== null` instead to make it clear that this is about a null check, not about avoiding instances of other classes.

Reply:
I'll do that, but: https://rskuipers.com/entry/from-assumptions-to-assertions